pkg/boot/uefi/uefi_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uefi
+
+import (
+	"testing"
+
+	"github.com/u-root/u-root/pkg/acpi"
+	"github.com/u-root/u-root/pkg/boot/kexec"
+)
+
+type kexecLoadFunc func(entry uintptr, segments kexec.Segments, flags uint64) error
+
+func TestLoadFvImage(t *testing.T) {
+	fv, err := New("testdata/uefi.fd")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func(old func() (*acpi.RSDP, error)) { getRSDP = old }(getRSDP)
+	getRSDP = func() (*acpi.RSDP, error) {
+		t.Log("mock acpi.GetRSDP()")
+		return &acpi.RSDP{}, nil
+	}
+	defer func(old func() (int64, int64, error)) { getSMBIOSBase = old }(getSMBIOSBase)
+	getSMBIOSBase = func() (int64, int64, error) {
+		t.Log("mock getSMBIOSBase()")
+		return 100, 200, nil
+	}
+	defer func(old kexecLoadFunc) { kexecLoad = old }(kexecLoad)
+	kexecLoad = func(entry uintptr, segments kexec.Segments, flags uint64) error {
```

Comment:
a real mock kexecLoad could/should test some kexec_load invariants: page-aligned segments, less than 16 segments, etc.

Reply:
Agreed. In that case, I will we should refactor the existing kexec pkg to have a mock kexec and other tests can include it. WDYT. I will first add a TODO here.